submissions/madmaxWMFU/Friends-App/js/app.js

```diff
@@ -0,0 +1,153 @@
+                        <div class="profile-card__name"><span>${user.name.last}</span></div>
+                        <div class="profile-card__name"><span>${user.location.city}, ${user.nat}</span></div>
+                        <div class="profile-card__name"><address>Place: ${user.location.street.name}, ${user.location.street.number}</address></div>
+                    </div> 
+                    <span class="user-card__age">${user.dob.age}</span>
+                </div>`;
+    friendZone.insertAdjacentHTML("beforeEnd", temp);
+}
+
+const cleanFriendsZone = () => friendZone.innerHTML = "";
+
+const showErrorMessage = () => friendZone.insertAdjacentHTML("beforeEnd", '<h4 class="friends-zone__error">Something went wrong...</h4>');
+
+const renderFriendsList = (allFriends) => allFriends.forEach(drawFriendsCards);
+
+const getFilterList = {
+    filterByGender: (list, {gender}) => {
+        if(gender !== "all") {
+          return list.filter((person) => person.gender === gender); 
+        } else {
+            return allFriends.currentList;
+        }
+    },
+    filterByAge: (list, {ageTill, ageTo}) => {
+        return list.filter((person) => ageTill <= person.dob.age && person.dob.age <= ageTo);
+    },
+    filterByName: (list, {name}) => {
+        if(name != "") {
+            return list.filter((person) => person.name.first.toLowerCase().includes(name));
+        } else {
+            return allFriends.changeList;
+        }
+    }
+}
+
+const getGenderValue = () => searchGender.find(item => item.checked === true).value;
+
+const applyFilters = () => {
+    Object.keys(getFilterList).forEach((filter) => {
+        allFriends.changeList = getFilterList[filter](allFriends.changeList, searchValues);
+    });
+}
+
+const renderFilterList = () => {
+    sorts.value = "none";
+    allFriends.changeList = [...allFriends.currentList];
+    searchValues.name = searchName.value;
+    searchValues.gender = getGenderValue();
+    searchValues.ageTill = searchAgeTill.value;
+    searchValues.ageTo = searchAgeTo.value;
+   
+    applyFilters();
+    cleanFriendsZone();
+    renderFriendsList(allFriends.changeList);  
+}
+
+const sortByName = (a, b) => a.name.first < b.name.first ? -1 : 1;
+
+const getSortByNameList = (option) => {
+    return allFriends.changeList.sort((firstPerson, secondPerson) => {
+        if(option === "nameAsc") {
+            return sortByName(firstPerson, secondPerson);
+        } else {
+            return sortByName(secondPerson, firstPerson);
+        }
+    });
+} 
+
+const sortByAge = (a, b) => a.dob.age - b.dob.age;
+
+const getSortByAgeList = (option) => {
+    return allFriends.changeList.sort((firstPerson, secondPerson) => {
+        if(option === "ageAsc") {
+            return sortByAge(firstPerson, secondPerson); 
+        } else {
+            return sortByAge(secondPerson, firstPerson);
+        }
+    });
+}
+
+const renderSortFriendList = ({target}) => {
+    if(target.value != "none")
```

Comment:
I see that in some cases you use braces and in some cases - not. This is inconsistence. Please, follow one codestyle.
Also, usually in JS we use braces, because it is easier to read code with them. It is not obligatory, but it is common used approach.

Reply:
In general, I tried to reduce as much code as possible, and I did not think that its readability suffers because of this.
I thought that if do not use the construction `else`, then I can't write curly brackets

Reply:
I see. Reducing amount of code by removing brackets is not very effective, just because you will not reduce many :)

Also, hope while this code works well with browser, because I haven't checked, it will provide potential bugs for [babel](https://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=GYVwdgxgLglg9mABACQKYBt1wOpwE7oAmAFAN4DmeqqsY5iAvIgEQAWGWzANIpdVKkKNEAcma4ChZiJ4BnGOlRgow4AEN0s1DwQBZOCC0B5AG6o8XAL4BKRKQBQ9xIhjBiAQj40Yda6SpQIHhIYCCYlgDcjs4xAPSxiAAqRgAiRgBciCkIIiqGqIh4amCEcAC2LkhUJeZOiIoqoRVMumpQrAB0wFj4iMSt7R1FJeXEtgBUiACMAKIA1ADs1h1QcADKUHg-5GNDqAAO6GoQqMSxADod54RzsTDkPMzM1nV1rsTyisqMDEybIKgXjFEABILy0ehMEQiKIxEEgz5KFRMdSaVCwuFNYQABiidQCQSQAB5CDATIgIEdZLIAHJqMqoBgiNCYHD4IgiRCwKCKBikAAGAE0DIg1FRECYYPJVnhEE0AEbmRAAElIcpAFUs_MsiD0BmMZjwfL1-VM5ksAD5ojEibJNghyBa1eDth1ZOgYCdiIhsTwpohlqsAKr7fbmADCai0Y0Qc14VG8dDdHq9U0DcAAMnAAO4RqOnWyWImxO14B1W4EUBMQjpKQiybAwdrEbjPRAAfhYXcytv2xUQdoAnrzSKQIHAsHhMiJznxByJLJbuCxi7I-2ALYhLHVnETUGUrSDndXBFv7CDi_uKzE1R8FEigcCOywOswdzFMsx3Mxt-_i6STAtPEdUQIA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=7.6.4&externalPlugins=).

As an example, code witch looks like this:

```
if(silent === true)
    	greeting = '';
   	silent = false;
    	num = 0;
```

After transpilation will look like this:

```
  if (silent === true) greeting = '';
  silent = false;
  num = 0;
```
Same behaviour with [prettier](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0AEAEnADYkQDqEATiQCYAUwA5tXHLlM-gLz4A6IABalyggDT5W7eHV74A5IKq06ghZIDOOEghjzUAQxKa4k6AFkI6UwHkAbnGriAvgEp8wflG-58OVAYAQmkOHC43YDYYdGoofCh0MhcAbm9fPz8Aeiz8ABVbABFbJHxC6AV9Gzh8akMoOggAW394tganDN19RJa+C0MYIQA6VHIafAYBoeG6huaGDwAqfABGAFEAagB2N2GYCABlGGpw5kXZuAAHEkMwOAYs-n5h57pNrJxmSUFBNwyMgEGNpdLBeDw+Cd0HB-vFMqFONw+AoFGk4X4QXoDMZTGjMgl0H18AAGNEZaKxeIAHjoOHs+DAt00mgAcoYmnAeApiGRKDR6Ap8LgYLoeMAAAYATWs+EMbHw9hw2gO1AJTQARk58AASYBq-AucUufCWax2RzUMWm6oOJwuAB86XR+CpmhO0GY9r1CLOw00JBw9wYJMkq3wewOAFUrlcnABhQymRb4TZSNhhLh+gNB1YRiAAGQgAHd44mHh4XFSsm7qB7Hc6WOnEcMEHRNBQcEMGBI-vgAPwCECD-ClV1Xer4N0AT1FwGAkHI1FKSn40inChcDokg6rmnHUHtBoyfipcCa9f4MG9Tbgchc6RgVbP9fxeuBOj0sPxfgHgleIGPTJSkEIJBHvHxnSrWl7HtMkoGNfAQHEEAICuXBoE0ZBQDlWsiwABTlBBMJQQx7AgHA6CQkB1TqMAAGsOEOccwDOZAoTMYQYCaEgKCETs4D3O44EOIjOzpTsp2QcBmSo8JTGoGA8LqZgmkMZAjBMDiACtNAADwAIVohiYEOdk4HzcI4HUnFtL0w4zl0ABFdAIHgazNOQ8dqHkqT1UMdUZ2gKirlOWAOzoIZkAADmJTza1MCg6iuKSQoEpxHCotgAEd0BwNglMMFS1KQDTTGQ0wmhwNjqGhcqHLgZzXKskqbOQmB-PCyKkAAJjauodDOONmlUqSoGgKzkOqPJ-OI0q4E3IA)

As you probably know, on a job, 99% of time you writing code, which will be transpiled. So it will produce bugs. 
Even when you run formatter like prettier. And this one one is especially bad, because formatter don't change existing logic and if this will happen... well, yikes. Very unpredictable. Imagine if this will happen on CI. You push your code into repo and somewhere there it will run prettier on it. So locally everything is fine, but on CI - it will fail during PR checks. Or event on production.

Anyway, I strongly suggest you not to use such practice outside your pet-projects and encourage you to make code more DRY. This will definitely will reduce amount of it :)